### PR TITLE
Pinecone: Fix stale vectors on starter pods

### PR DESCRIPTION
**Pull request by flash1293**
## What

On starter pods, the connector is deleting documents by querying them via search, then deleting by id. In the previous version, it would only do a single search for 10k documents. That means if there are more than 10k vectors in the index, the search might not return all vectors which means stale vectors will stick around. This PR changes the logic to query-and-delete until there are no vectors returned anymore.

This problem only affects starter pods as production pods can delete by filter directly.

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  destination-pinecone test report (commit [`3ff432285f`](https://github.com/airbytehq/airbyte/pull/30718/commits/3ff432285fc1d72108a6ec4596d9fdca1a5fb6bb)) - ✅

⏲️  Total pipeline duration: 01mn46s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Connector package install                                                   | ✅        |
| Build destination-pinecone docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                  | ✅        |
| Integration tests                                                           | ✅        |
| Acceptance tests                                                            | ✅        |
| Code format checks                                                          | ✅        |
| Validate airbyte-integrations/connectors/destination-pinecone/metadata.yaml | ✅        |
| Connector version semver check                                              | ✅        |
| Connector version increment check                                           | ✅        |
| QA checks                                                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_pinecone-fix/1695631623/3ff432285fc1d72108a6ec4596d9fdca1a5fb6bb/destination-pinecone/0.0.12/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:3ff432285fc1d72108a6ec4596d9fdca1a5fb6bb)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-pinecone test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-pinecone test report (commit [`0f1ab49690`](https://github.com/airbytehq/airbyte/pull/30718/commits/0f1ab4969037c4634ef31fd974c05703df9bb34b)) - ✅

⏲️  Total pipeline duration: 02mn07s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Connector package install                                                   | ✅        |
| Build destination-pinecone docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                  | ✅        |
| Integration tests                                                           | ✅        |
| Acceptance tests                                                            | ✅        |
| Code format checks                                                          | ✅        |
| Validate airbyte-integrations/connectors/destination-pinecone/metadata.yaml | ✅        |
| Connector version semver check                                              | ✅        |
| Connector version increment check                                           | ✅        |
| QA checks                                                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_pinecone-fix/1695716363/0f1ab4969037c4634ef31fd974c05703df9bb34b/destination-pinecone/0.0.12/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:0f1ab4969037c4634ef31fd974c05703df9bb34b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-pinecone test
```

File: airbyte-integrations/connectors/destination-pinecone/destination_pinecone/indexer.py [hunk 1]
```diff
@@ -54,12 +54,14 @@ def delete_vectors(self, filter):
...
-            batches = create_chunks(vector_ids, batch_size=MAX_IDS_PER_DELETE)
-            for batch in batches:
-                self.pinecone_index.delete(ids=list(batch))
+        while len(query_result.matches) > 0:
+            vector_ids = [doc.id for doc in query_result.matches]
+            if len(vector_ids) > 0:
+                # split into chunks of 1000 ids to avoid id limit
+                batches = create_chunks(vector_ids, batch_size=MAX_IDS_PER_DELETE)
+                for batch in batches:
+                    self.pinecone_index.delete(ids=list(batch))
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-pinecone/destination_pinecone/indexer.py:63]: Just for my info, are we expecting to have to run through multiple queries? I see there's a loop through batches that should mitigate the 1000 id limit - but in that case, I'm not sure if there's also a limit applied to the query.

**Review by aaronsteers — APPROVED** [review]
One question added inline but no issues found.

**Pull request merged**